app/uk/gov/hmrc/apisubscriptionfields/repository/SubscriptionFieldsIdRepository.scala

```diff
@@ -17,40 +17,40 @@
 package uk.gov.hmrc.apisubscriptionfields.repository
 
 import java.util.UUID
+import javax.inject.{Inject, Singleton}
 
+import com.google.inject.ImplementedBy
 import play.api.libs.json._
 import play.modules.reactivemongo.MongoDbConnection
 import reactivemongo.api.DB
 import reactivemongo.api.commands.WriteResult
-import reactivemongo.api.indexes.IndexType.Ascending
 import reactivemongo.api.indexes.Index
+import reactivemongo.api.indexes.IndexType.Ascending
 import reactivemongo.bson.{BSONDocument, BSONObjectID}
-import uk.gov.hmrc.apisubscriptionfields.model.{ApiSubscription, MongoFormatters}
+import uk.gov.hmrc.mongo.ReactiveRepository
 import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
-import uk.gov.hmrc.mongo.{ReactiveRepository, Repository}
 
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.Future
 
-trait SubscriptionFieldsIdRepository extends Repository[ApiSubscription, BSONObjectID] {
+@ImplementedBy(classOf[SubscriptionFieldsIdMongoRepository])
+trait SubscriptionFieldsIdRepository {
 
-  def save(subscription: ApiSubscription): Future[Unit]
+  def save(subscription: SubscriptionFields): Future[Unit]
 
-  def fetchById(id: String): Future[Option[ApiSubscription]]
-  def fetchByFieldsId(fieldsId: UUID): Future[Option[ApiSubscription]]
+  def fetchById(id: String): Future[Option[SubscriptionFields]]
+  def fetchByFieldsId(fieldsId: UUID): Future[Option[SubscriptionFields]]
 
   def delete(id: String): Future[Unit]
 }
 
-class SubscriptionFieldsIdMongoRepository(implicit mongo: () => DB)
-  extends ReactiveRepository[ApiSubscription, BSONObjectID]("subscriptionFieldsId", mongo,
-    MongoFormatters.formatApiSubscription, ReactiveMongoFormats.objectIdFormats)
+@Singleton
+class SubscriptionFieldsIdMongoRepository @Inject() (mongoDbProvider: MongoDbProvider)
+  extends ReactiveRepository[SubscriptionFields, BSONObjectID]("subscriptionFieldsId", mongoDbProvider.mongo,
```

Comment:
The mongo database takes the name from the microservice: `api-subscription-fields`.
The mongo collection is called here `subscriptionFieldsId`. I suggest to call it `subscriptionFields` (in fact we save both the id and the fields, not just the id)

Reply:
Done.